Sources/Tools/MaxMain/MaxCompat.h

```diff
@@ -68,8 +72,18 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
 
 #   define ENUMDEPENDENTS(maxObject, proc) \
         maxObject->DoEnumDependents(proc);
+
+#   define INIT_CUSTOM_CONTROLS(instance)
 #endif //MAX_VERSION_MAJOR
 
+#if MAX_VERSION_MAJOR <= 10 // Max 2008
+#   define GETNAME_RETURN_TYPE TCHAR*
+#   define SETTEXT_VALUE_TYPE MCHAR*
+#else
+#   define GETNAME_RETURN_TYPE const TCHAR*
+#   define SETTEXT_VALUE_TYPE const MCHAR*
```

Comment:
I'm not a fan of these `*_RETURN_TYPE` and `*_VALUE_TYPE` macros, mostly because they completely hide the actual type for the sake of a simple `const`.  Why not instead introduce a `MAX10_CONST`, `MAX12_CONST` etc for places where various versions of the SDK changed types to `const`?

Reply:
I suspect my thought process here was future-proofing type changes outside of const-ness, but your solution seems more flexible and readable. 😃